feat: 添加 ansible roles

YkeLit opened the pull request:
在 docker compose 环境下 部署以下应用
- prometheus
- blackbox exporter
- grafana

YkeLit committed:
feat: 添加 ansible roles

在 docker compose 环境下 不是以下应用
- prometheus
- blackbox exporter
- grafana

YkeLit commented:
## 这个 roles 还差的组件：
- alert 相关
- node exporter
## 现在是初步设计：
- docker 环境使用 `docker compose` 部署
- k8s 环境使用 `helm`部署, 后期添加上自定义  `values` 文件
## 目前的一些设计上的问题：
-  每个 roles 上都有一个判断当前环境的 tasks 是否重复

fahchen commented on ansible/roles/blackbox-exporter/defaults/main.yml:
```diff
@@ -0,0 +1,26 @@
+      valid_status_codes: []
+      preferred_ip_protocol: "ip4"
+      fail_if_body_not_matches_regexp:
+        - "\"code\":\\s*1,"
```
<img width="297" alt="image" src="https://user-images.githubusercontent.com/855678/233916675-7ffcbb6f-aa79-4d32-85b2-e7788252225a.png">
你编辑器改下配置吧，这个问题出现很多次了

c01dcat replied:
可以使用一下 [ansible-lint](https://github.com/ansible/ansible-lint)，我运行了一下发现很多空格和新行上的提示
<img width="635" alt="截屏2023-04-24 17 22 18" src="https://user-images.githubusercontent.com/2656694/233955771-f2f6c780-b574-4dfd-8ee1-f77bdb572a3a.png">

YkeLit committed:
feat: 添加 ansible roles

在 docker compose 环境下 部署以下应用
- prometheus
- blackbox exporter
- grafana
- alertmanager and prometheus-alert
- node exporter

c01dcat commented on ansible/roles/alertmanager/templates/prometheus-alert.conf.j2:
```diff
@@ -0,0 +1,250 @@
+#登录用户名
+login_user=prometheus
+#登录密码
+login_password=WW5sNllXNTBaV0ZPQo
```
密码，token 和 api_key 等敏感信息如果有必要可以用 [Vault](https://docs.ansible.com/ansible/latest/vault_guide/index.html)

YkeLit committed:
style: 格式修改

YkeLit committed:
feat: 添加 cAdvisor 部署

YkeLit committed:
feat: 添加 cAdvisor 部署

YkeLit committed:
feat: 添加 resolve_timeout 设置

YkeLit committed:
Merge commit 'FETCH_HEAD' into feat/yangke/prometheus-ansible
